File: AOC 2022/Python drafts/Day2Part1.py
#pre-processing
file = open("aoc22day2.txt",'r')
data = file.read()
file.close()

round = data.split("\n")
decisions = []
for r in round:
  decisions.append(r.split(" "))

#decisions = [["C","X"],["A","Y"],["B","X"],["C","Z"]]
#solution
ref = ['X','A','Y','B','Z','C']
points = 0

for round in decisions:
  opp, plyr = round
  o, p = ref.index(opp), ref.index(plyr)
  if (o == 5 and p == 0):
    #avoid winning edge case
    points += 6
  elif (p-o)==1:
    #win
    points += 6
  elif (o-p)==1:
    #draw
    points += 3
  
  if p == 0:
    points += 1
  elif p==2:
    points +=2
  else:
    points += 3
    
print(points)

File: AOC 2022/C++ solutions/Day2Part1.cpp
#include <iostream> 
#include <fstream>
#include <algorithm>
using namespace  std;

int main() {
  #pre-processing and init
  string s;
  char ref[] = {'X','A','Y','B','Z','C'};
  int points = 0;
  ifstream dataFile("aoc22day2.txt");
  
  //still incomplete due to incorrect answer
  while (getline(dataFile, s)) {//each line is a new round
    char opp, plyr = s[0], s[2]; //label char and index of value in ref array
    auto it = find(begin(ref),end(ref),opp); 
    int o = it - begin(ref);
    it = find(begin(ref),end(ref),plyr);
    int p = it - begin(ref);

    //mild bruteforce method
    if (o==5 && p==0){ //edge winning case if input is last and first value
      points += 6;
    }
    else if (p-o==1){ //win condition (according to ref array)
      points += 6;
    }
    else if (o-p==1){ //draw condition
      points += 3;
    }
    //otherwise will be a loss (0 points) - no change

    if (p==0){ //if rock
      points += 1;
    }
    else if (p==2){ //if paper
      points += 2;
    }
    else{ //otherwise is scissors
      points += 3;
    }
  }

  //print final points
  cout << points << endl;
}

